packages/happn-cluster/lib/services/orchestrator.js

```diff
@@ -1,504 +1,402 @@
-    reports.forEach(function (line) {
-      _this.log.debug(line);
-    });
-    this.log.debug('');
+  startIntervals() {
+    this.intervals.health = setInterval(
+      this.healthReport.bind(this),
+      this.config.timing.healthReport
+    );
+    this.keepAlive();
+    this.intervals.keepAlive = setInterval(this.keepAlive.bind(this), this.config.timing.keepAlive);
   }
 
-  if (unstableCount !== 0) return;
-
-  clearInterval(this.stableReportInterval);
-
-  if (this.stabiliseWaiting.length === 0) return;
-
-  peerCount = Object.keys(this.peers).length;
-  if (peerCount < this.config.minimumPeers) {
-    if (!this.stableAwaitingMinimumPeers) {
-      this.log.warn('requires %d more peers to stabilise', this.config.minimumPeers - peerCount);
+  stabilised(callback) {
+    if (this.stableTimeout) return;
+    if (typeof callback === 'function') this.stabiliseWaiting.push(callback);
+    if (this.config.timing.stabiliseTimeout) {
+      this.stableTimeout = setTimeout(() => {
+        var error = new Error('failed to stabilise in time');
+        error.name = 'StabiliseTimeout';
+        this.doStabilisedCallbacks(error);
+      }, this.config.timing.stabiliseTimeout);
     }
-    this.stableAwaitingMinimumPeers = true;
-    return;
+    this.__stateUpdate();
   }
 
-  this.stableAwaitingMinimumPeers = false;
-
-  this.log.info('stabilised');
-  clearInterval(this.stableTimeout);
-
-  while ((callback = this.stabiliseWaiting.shift()) !== undefined) callback();
-};
-
-Orchestrator.prototype.__addPeer = function (member) {
-  if (this.peers[member.name]) return;
-
-  this.peers[member.name] = member; // includes self by name
-  if (member.self) property(this.peers, '__self', member); // non enumerable __self
-  this.emit('peer/add', member.name, member);
-
-  if (this.stableAwaitingMinimumPeers) {
-    this.log.info(
-      'cluster size %d/%d (%s arrived)',
-      Object.keys(this.peers).length,
-      this.config.minimumPeers,
-      member.name
-    );
-    return;
+  doStabilisedCallbacks(error) {
+    error = Array.isArray(error) ? error[0] : error;
+    let callback;
+    while ((callback = this.stabiliseWaiting.shift()) !== undefined) callback(error);
   }
-  this.log.info('cluster size %d (%s arrived)', Object.keys(this.peers).length, member.name);
-};
 
-Orchestrator.prototype.removePeer = function (member) {
-  if (!this.peers[member.name]) return;
-  delete this.peers[member.name];
-  if (member.self) delete this.peers.__self;
-  this.emit('peer/remove', member.name, member);
-  this.log.info(`[CLUSTER_MEMBERSHIP]: orchestrator removePeer ${member.name}`);
-  if (this.stableAwaitingMinimumPeers) {
-    this.log.info(
-      'cluster size %d/%d (%s left)',
-      Object.keys(this.peers).length,
-      this.config.minimumPeers,
-      member.name
-    );
-    return;
+  async stop(opts, cb) {
+    if (typeof opts === 'function') cb = opts;
+    for (let interval of Object.values(this.intervals)) {
```

Comment:
Object.values(this.intervals).forEach(clearInterval)

Reply:
done